lib/http2/connection.c

```diff
@@ -401,7 +401,8 @@ static ssize_t expect_continuation_of_headers(h2o_http2_conn_t *conn, const uint
             conn->_read_expect = expect_default;
             if (stream->state == H2O_HTTP2_STREAM_STATE_RECV_HEADERS) {
                 hret = handle_incoming_request(conn, stream, (const uint8_t *)conn->_headers_unparsed->bytes,
-                                               conn->_headers_unparsed->size, err_desc);
+                                               conn->_headers_unparsed->size, (frame.flags & H2O_HTTP2_FRAME_FLAG_END_STREAM),
```

Comment:
Looking at this line, it seems that the PR is based on the assumption that the CONTINUATION frame carries the END_STREAM flag.

However, the specification states that the HEADERS frame carries the END_STREAM flag even if CONTINUATION frames were used.

> A HEADERS frame carries the END_STREAM flag that signals the end of a stream. However, a HEADERS frame with the END_STREAM flag set can be followed by CONTINUATION frames on the same stream. Logically, the CONTINUATION frames are part of the HEADERS frame. 
> ([section 8.1](https://http2.github.io/http2-spec/#rfc.section.8.1), RFC 7540)

Therefore I believe that the current approach is doing what is defined in the specification.